MobiFlight/Joysticks/Joystick.cs

```diff
@@ -152,6 +120,69 @@ protected virtual void EnumerateDevices()
+            string axisLabel = "Unknown";
+            try
+            {
+                var OffsetAxisName = GetAxisNameForUsage(device.Usage);
+                axisName = $"{AxisPrefix} {OffsetAxisName}";
+                axisLabel = MapDeviceNameToLabel($"{AxisPrefix} {OffsetAxisName}");
+
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                LogAddInputError(device, axisLabel, "Axis can't be mapped");
+                return;
+            }
+            Axes.Add(new JoystickDevice() { Name = axisName, Label = axisLabel, Type = DeviceType.AnalogInput, JoystickDeviceType = JoystickDeviceType.Axis });
+            LogInputAdded(device, axisLabel);
+        }
+
+        protected virtual void RegisterButton(DeviceObjectInstance device)
+        {
+            // Use the device.Usage value so this is consistent with how Axes are referenced and avoid ID collisions
+            // when looking up names in the the .joystick.json file.
+            var buttonName = $"{ButtonPrefix} {device.Usage}";
+            var buttonLabel = MapDeviceNameToLabel(buttonName);
+            Buttons.Add(new JoystickDevice() { Name = buttonName, Label = buttonLabel, Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.Button });
+            LogInputAdded(device, buttonLabel);
+        }
+
+        protected virtual void RegisterPOV(DeviceObjectInstance device)
+        {
+            String name = device.Name;
+
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}U", Label = $"{name} (↑)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}UR", Label = $"{name} (↗)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}R", Label = $"{name} (→)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}DR", Label = $"{name} (↘)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}D", Label = $"{name} (↓)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}DL", Label = $"{name} (↙)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}L", Label = $"{name} (←)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            POV.Add(new JoystickDevice() { Name = $"{PovPrefix} {name}UL", Label = $"{name} (↖)", Type = DeviceType.Button, JoystickDeviceType = JoystickDeviceType.POV });
+            LogInputAdded(device, name);
+        }
+
+        protected void LogInputAdded(DeviceObjectInstance device, string label)
```

Comment:
Trading debug information (calling function name) for DRYness here. If having the log point to dedicated logging methods is unwanted, I have a different compromise in mind as an alternative option (generate the string in a method and call Log.Instance.log with that).